compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.cpp

```diff
@@ -65,26 +66,225 @@ static FailureOr<AffineMap> getComposedAffineMap(Attribute attr) {
   Builder b(ctx);
   auto opTypeAttr = EncodingOpTypeAttr::get(ctx, opType);
   auto mapsAttr = maps.empty() ? ArrayAttr() : b.getAffineMapArrayAttr(maps);
-  auto roundDimsToAttr = roundDimsTo.empty()
-                             ? DenseI64ArrayAttr()
-                             : b.getDenseI64ArrayAttr(roundDimsTo);
+  auto iterationSizesAttr =
+      iterationSizes.empty() ? ArrayAttr() : b.getI64ArrayAttr(iterationSizes);
   auto layoutsAttr = layouts.empty() ? ArrayAttr() : b.getArrayAttr(layouts);
   return get(ctx, b.getIndexAttr(operandIndex), opTypeAttr,
-             b.getTypeArrayAttr(elemTypes), mapsAttr, roundDimsToAttr,
+             b.getTypeArrayAttr(elemTypes), mapsAttr, iterationSizesAttr,
              layoutsAttr);
 }
 
+/// Parse a comma-separated list of key-value pairs with a specified
+/// delimiter. This performs similar parsing as the the assembly format
+/// `struct` directive parser with a specified delimiter. The variables are
+/// printed in the order they are specified in the argument list but can be
+/// parsed in any order.
+///
+/// Example:
+/// <
+///   foo = something_parsed_by_a_custom_parser,
+///   bar = something_parsed_by_a_different_custom_parser,
+///   ...
+/// >
+static ParseResult
+parseStruct(AsmParser &p, AsmParser::Delimiter delimiter,
+            ArrayRef<StringRef> keywords,
+            ArrayRef<llvm::function_ref<ParseResult()>> parseFuncs) {
+  assert(keywords.size() == parseFuncs.size());
+  auto keyError = [&]() -> ParseResult {
+    InFlightDiagnostic parseError =
+        p.emitError(p.getCurrentLocation(), "expected one of: ");
+    llvm::interleaveComma(keywords, parseError, [&](StringRef kw) {
+      parseError << '`' << kw << '`';
+    });
+    return parseError;
+  };
+  SmallVector<bool> seen(keywords.size(), false);
+  DenseMap<StringRef, size_t> keywordToIndex;
+  for (auto &&[idx, keyword] : llvm::enumerate(keywords))
+    keywordToIndex[keyword] = idx;
+  return p.parseCommaSeparatedList(
+      delimiter,
+      [&]() -> ParseResult {
+        StringRef keyword;
+        if (failed(p.parseOptionalKeyword(&keyword)))
+          return keyError();
+        if (!keywordToIndex.contains(keyword))
+          return keyError();
+        size_t idx = keywordToIndex[keyword];
+        if (seen[idx]) {
+          return p.emitError(p.getCurrentLocation(), "duplicated `")
+                 << keyword << "` entry";
+        }
+        if (failed(p.parseEqual()))
+          return failure();
+        if (failed(parseFuncs[idx]()))
+          return failure();
+        seen[idx] = true;
+        return success();
+      },
+      "parse struct");
+}
+
+/// Utility to parse an array of integer and/or dynamic values (`?`).
+static FailureOr<ArrayAttr> parseDynamicI64ArrayAttr(AsmParser &p) {
+  SmallVector<Attribute> integerVals;
+  if (failed(p.parseLSquare()))
+    return failure();
+  if (failed(p.parseCommaSeparatedList([&] {
+        int64_t value{ShapedType::kDynamic};
```

Comment:
nit: do not intialize an int like this. please use `=` assignment.

```suggestion
        int64_t value = ShapedType::kDynamic;
```

https://llvm.org/docs/CodingStandards.html#do-not-use-braced-initializer-lists-to-call-a-constructor

Reply:
Thanks, updated it.